lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -584,20 +593,263 @@ impl<'s> SegmentHolder {
+                    .into_sparse_vector_data()
+                    .map_err(|err| OperationError::service_error(format!("Failed to source sparse vector configuration from collection parameters: {err:?}")))?,
+                payload_storage_type: collection_params.payload_storage_type(),
+            },
+            // Base config on existing appendable or non-appendable segment
+            None => {
+                segments_lock
+                    .random_appendable_segment()
+                    .ok_or_else(|| OperationError::service_error("No existing segment to source temporary segment configuration from"))?
+                    .get()
+                    .read()
+                    .config()
+                    .clone()
+            }
+        };
+
+        // Create temporary appendable segment to direct all proxy writes into
+        let tmp_segment = LockedSegment::new(build_segment(collection_path, &config, false)?);
+
+        // List all segments we want to snapshot
+        let segment_ids = segments_lock
+            .segments
+            .keys()
+            .copied()
+            .collect::<Vec<SegmentId>>();
+
+        // Create proxy for all segments
+        let mut new_proxies = Vec::with_capacity(segment_ids.len());
+        for segment_id in segment_ids {
+            let segment = segments_lock.segments.get(&segment_id).unwrap();
+            let mut proxy = ProxySegment::new(
+                segment.clone(),
+                tmp_segment.clone(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+            );
+
+            // Write segment is fresh, so it has no operations
+            // Operation with number 0 will be applied
+            proxy.replicate_field_indexes(0)?;
+            new_proxies.push((segment_id, proxy));
+        }
+
+        // Save segment version once all payload indices have been converted
+        // If this ends up not being saved due to a crash, the segment will not be used
+        match &tmp_segment {
+            LockedSegment::Original(segment) => {
+                let segment_path = &segment.read().current_path;
+                SegmentVersion::save(segment_path)?;
+            }
+            LockedSegment::Proxy(_) => unreachable!(),
+        }
+
+        // Replace all segments with proxies
+        // We cannot fail past this point to prevent only having some segments proxified
+        let mut proxies = Vec::with_capacity(new_proxies.len());
+        let mut write_segments = RwLockUpgradableReadGuard::upgrade(segments_lock);
+        for (segment_id, mut proxy) in new_proxies {
+            // Replicate field indexes a the second time, because optimized segments could have
+            // been changed. The probability is small, though, so we can afford this operation
+            // under the full collection write lock
+            let op_num = proxy.version() + 1;
```

Comment:
It might not matte in this case, but I would prefer to only increase op_num in WAL. This should guarantee that version number won't accumulate independently from the WAL, therefore we would guarantee that new upcomming operations always have newer version number

Reply:
Good suggestion, I'll remove the +1's.

Reply:
I see you already did so. Thanks!